### PR TITLE
refactor: replace memoize with django utils

### PR DESCRIPTION
**Pull request by anonrig**
Let's remove internal implementation in favor of `functools`

Ref: https://github.com/getsentry/team-ingest/issues/400

### CODE REVIEW

File: src/sentry/utils/cache.py [hunk 1]
```diff
@@ -1,44 +1,14 @@
...
 
-__all__ = ["cache", "memoize", "default_cache", "cache_key_for_event"]
+__all__ = ["cache", "default_cache", "cache_key_for_event"]
 
 default_cache = cache
 
 T = TypeVar("T")
 
 
-class memoize(Generic[T]):
```
Comment (asottile-sentry) [inline src/sentry/utils/cache.py:13]: I think you'll have to do `getsentry` first -- it looks like it uses this as well

Comment (anonrig) [reply src/sentry/utils/cache.py:13]: True - https://github.com/getsentry/getsentry/pull/13419

**Review by asottile-sentry — APPROVED** [review]

**Review by asottile-sentry — APPROVED** [review]

**Review by armenzg — APPROVED** [review]

**Pull request merged**